src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,199 @@
+        )
+
+        # Canny edge detect
+        yield from bps.abs_set(
+            self.oav.canny_lower_threshold_pv,
+            self.oav_parameters.canny_edge_lower_threshold,
+        )
+        yield from bps.abs_set(
+            self.oav.canny_upper_threshold_pv,
+            self.oav_parameters.canny_edge_upper_threshold,
+        )
+        # "Close" morphological operation
+        yield from bps.abs_set(self.oav.close_ksize_pv, self.oav_parameters.close_ksize)
+
+        # Sample detection
+        yield from bps.abs_set(
+            self.oav.sample_detection_scan_direction_pv, self.oav_parameters.direction
+        )
+        yield from bps.abs_set(
+            self.oav.sample_detection_min_tip_height_pv,
+            self.oav_parameters.minimum_height,
+        )
+
+        # Connect MXSC output to MJPG input
+        yield from self.start_mxsc(
+            self.oav_parameters.input_plugin + "." + self.oav_parameters.mxsc_input,
+            self.oav_parameters.min_callback_time,
+            self.oav_parameters.filename,
+        )
+
+        yield from bps.abs_set(
+            self.oav.input_pv, self.oav_parameters.input_plugin + ".MXSC"
+        )
+
+        # zoom is an integer value, stored as a float in a string so
+        # we may need a .0 on the end? GDA does this with
+        # zoomString = '%1.0dx' % float(zoom)
+        # which seems suspicious, we may want to think about doing this in a nicer way
+        yield from bps.abs_set(
+            self.oav_camera.zoom, f"{float(int(self.oav_parameters.zoom))}x", wait=True
+        )
+
+        if (yield from bps.rd(self.oav_backlight.control)) == "Out":
+            yield from bps.abs_set(self.oav_backlight.control, "In", wait=True)
+
+        """
+       TODO: currently can't find the backlight brightness
+        this will need to be set up after issue
+        https://github.com/DiamondLightSource/python-artemis/issues/317
+        is solved
+         brightnessToUse = self.oav_parameters.extract_dict_parameter(
+            context, "brightness"
+        )
+
+        blbrightness.moveTo(brightnessToUse)
+        """
+
+    def start_mxsc(self, input_plugin, min_callback_time, filename=None):
+        yield from bps.abs_set(self.oav.input_plugin_pv, input_plugin)
+        yield from bps.abs_set(self.oav.enable_callbacks_pv, 1)
+        yield from bps.abs_set(self.oav.min_callback_time_pv, min_callback_time)
+        yield from bps.abs_set(self.oav.blocking_callbacks_pv, 0)
+
+        # I03-323
+        if filename is not None:
+            yield from bps.abs_set(self.oav.py_filename_pv, filename, wait=True)
+            yield from bps.abs_set(self.oav.read_file_pv, 1)
+
+        # Image annotations
+        yield from bps.abs_set(self.oav.draw_tip_pv, True)
+        yield from bps.abs_set(self.oav.draw_edges_pv, True)
+
+        # Image to send downstream
+        OUTPUT_ORIGINAL = 0
+        yield from bps.abs_set(self.oav.output_array_pv, OUTPUT_ORIGINAL)
```

Comment:
Should: I'm not sure what this means and the comment doesn't help much

Reply:
Tbh I'm not entirely sure. I'll ask Mark Williams.